scenario/elasticmoveapp.go

```diff
@@ -0,0 +1,107 @@
+	}
+)
+
+// UnmarshalJSON unmarshals ElasticMoveAppSettings from JSON
+func (settings *ElasticMoveAppSettings) UnmarshalJSON(arg []byte) error {
+	var actionCore DestinationSpace
+	if err := jsonit.Unmarshal(arg, &actionCore); err != nil {
+		return errors.Wrapf(err, "failed to unmarshal action<%s>", ActionElasticMoveApp)
+	}
+	var appSelectCore session.AppSelection
+	if err := jsonit.Unmarshal(arg, &appSelectCore); err != nil {
+		return errors.Wrapf(err, "failed to unmarshal action<%s>", ActionElasticMoveApp)
+	}
+	(*settings).DestinationSpace = actionCore
+	(*settings).AppSelection = appSelectCore
+	return nil
+}
+
+// Validate action (Implements ActionSettings interface)
+func (settings ElasticMoveAppSettings) Validate() error {
+	if (settings.DestinationSpaceId == "") == (settings.DestinationSpaceName == "") {
+		return errors.New("either specify DestinationSpaceId or DestinationSpaceName")
+	}
+	return nil
+}
+
+// Execute action (Implements ActionSettings interface)
+func (settings ElasticMoveAppSettings) Execute(sessionState *session.State, actionState *action.State, connection *connection.ConnectionSettings, label string, reset func()) {
+	host, err := connection.GetRestUrl()
+	if err != nil {
+		actionState.AddErrors(err)
+		return
+	}
+
+	entry, err := settings.AppSelection.Select(sessionState)
+	if err != nil {
+		actionState.AddErrors(errors.Wrap(err, "failed to perform app selection"))
+		return
+	}
+
+	destSpace, err := settings.ResolveDestinationSpace(sessionState)
+	if err != nil {
+		actionState.AddErrors(err)
+		return
+	}
+
+	spaceReference := elasticstructs.SpaceReference{SpaceID: destSpace.ID}
+	spaceReferenceJson, err := json.Marshal(spaceReference)
+	if err != nil {
+		actionState.AddErrors(err)
+		return
+	}
+
+	putApp := session.RestRequest{
+		Method:      session.PUT,
+		ContentType: "application/octet-stream",
+		Destination: fmt.Sprintf("%s/api/v1/apps/%s/space", host, entry.GUID),
+		Content:     spaceReferenceJson,
+	}
+
+	restHandler := sessionState.Rest
+	restHandler.QueueRequest(actionState, true, &putApp, sessionState.LogEntry)
+	if sessionState.Wait(actionState) {
+		actionState.AddErrors(errors.New("failed during app move"))
+		return
+	}
+	if putApp.ResponseStatusCode != http.StatusOK {
+		actionState.AddErrors(errors.Errorf("unexpected response code <%d> when putting app in new space: %s", putApp.ResponseStatusCode, putApp.ResponseBody))
+	}
+}
+
+func (settings DestinationSpace) ResolveDestinationSpace(sessionState *session.State) (*elasticstructs.Space, error) {
+	var moveToSpace *elasticstructs.Space
+	var err error
+	if settings.DestinationSpaceId != "" {
+		moveToSpace, err = sessionState.ArtifactMap.GetSpaceByID(settings.DestinationSpaceId)
+	} else if settings.DestinationSpaceName != "" {
+		moveToSpace, err = sessionState.ArtifactMap.GetSpaceByName(settings.DestinationSpaceName)
```

Comment:
This would require a preceeding elasticexplore action where the space has been looked up. We should probably use the function `searchForSpaceByName` from elastiexplore instead, this will look in artifactmap first, otherwise search for it.